app/uk/gov/hmrc/softdrinksindustrylevy/services/SdilPersistence.scala

```diff
@@ -17,180 +17,154 @@
 import uk.gov.hmrc.softdrinksindustrylevy.models._
-import com.google.inject.{ImplementedBy, Inject, Singleton}
-import play.modules.reactivemongo.ReactiveMongoComponent
+import com.google.inject.{Inject, Singleton}
+import org.mongodb.scala.model.{Filters, FindOneAndReplaceOptions, IndexModel, IndexOptions, Indexes, ReturnDocument, Updates}
+import uk.gov.hmrc.mongo.MongoComponent
+import uk.gov.hmrc.mongo.play.json.PlayMongoRepository
+import uk.gov.hmrc.softdrinksindustrylevy.services.SubscriptionWrap._
 
 import java.time._
 import scala.concurrent.{ExecutionContext => EC, _}
+import uk.gov.hmrc.softdrinksindustrylevy.models.json.internal._
+import play.api.libs.functional.syntax._
+import uk.gov.hmrc.softdrinksindustrylevy.services.ReturnsWrapper.returnsWrapperFormat
+
+case class SubscriptionWrap(
+  utr: String,
+  subscription: Subscription,
+  retrievalTime: LocalDateTime = LocalDateTime.now()
+)
+object SubscriptionWrap {
+
+  val subsWrapperFormat = Json.format[SubscriptionWrap]
+}
 
-@ImplementedBy(classOf[SdilMongoPersistence])
-trait SdilPersistence {
-
-  protected trait DAO[U, K, V] {
-    def update(user: U, key: K, value: V)(implicit ec: EC): Future[Unit]
-    def get(user: U, key: K)(implicit ec: EC): Future[Option[(V, Option[BSONObjectID])]]
-    def apply(user: U, key: K)(implicit ec: EC): Future[V] =
-      get(user, key).map { _.get._1 }
-    def list(user: U)(implicit ec: EC): Future[Map[K, V]]
-    def listVariable(user: U)(implicit ec: EC): Future[Map[K, V]]
-    def dropCollection(implicit ec: EC): Future[Boolean]
-
-    protected case class ReturnsWrapper(
-      utr: String,
-      period: ReturnPeriod,
-      sdilReturn: SdilReturn,
-      _id: Option[BSONObjectID] = None)
-
-    implicit val returnsFormatWrapper = Json.format[ReturnsWrapper]
-
-    def returnsMongo: ReactiveRepository[ReturnsWrapper, BSONObjectID]
-  }
-
-  def returns: DAO[String, ReturnPeriod, SdilReturn]
-
-  protected trait SubsDAO[K, V] {
-    def insert(key: K, value: V)(implicit ec: EC): Future[Unit]
-    def list(key: K)(implicit ec: EC): Future[List[V]]
-
-    protected case class SubscriptionWrapper(
-      utr: String,
-      subscription: Subscription,
-      retrievalTime: LocalDateTime = LocalDateTime.now(),
-      _id: Option[BSONObjectID] = None
-    )
-
-    import json.internal._
-    implicit val subscriptionsFormatWrapper = Json.format[SubscriptionWrapper]
-
-    def subscriptionsMongo: ReactiveRepository[SubscriptionWrapper, BSONObjectID]
-  }
+case class ReturnsWrapper(
+  utr: String,
+  period: ReturnPeriod,
+  sdilReturn: SdilReturn
+)
 
-  def subscriptions: SubsDAO[String, Subscription]
+object ReturnsWrapper {
+  val returnsWrapperFormat = Json.format[ReturnsWrapper]
 }
 
 @Singleton
-class SdilMongoPersistence @Inject()(mc: ReactiveMongoComponent) extends SdilPersistence {
-
-  val subscriptions = new SubsDAO[String, Subscription] {
+class SdilMongoPersistence @Inject()(
```

Comment:
Since you have separated the ReturnsPersistence, i think this class can be renamed as subscriptionPersistence? what do you think?